test/framework/filetools.py

```diff
@@ -1551,6 +1551,89 @@ def test_copy_files(self):
         error_pattern = "/toy-0.0.eb exists but is not a directory"
         self.assertErrorRegex(EasyBuildError, error_pattern, ft.copy_files, [bzip2_ec], copied_toy_ec)
 
+        # by default copy_files allows empty input list, but if allow_empty=False then an error is raised
+        ft.copy_files([], self.test_prefix)
+        error_pattern = 'One of more files to copy should be specified!'
+        self.assertErrorRegex(EasyBuildError, error_pattern, ft.copy_files, [], self.test_prefix, allow_empty=False)
+
+        # test special case: copying a single file to a file target via target_single_file=True
+        target = os.path.join(self.test_prefix, 'target')
+        self.assertFalse(os.path.exists(target))
+        ft.copy_files([toy_ec], target, target_single_file=True)
+        self.assertTrue(os.path.exists(target))
+        self.assertTrue(os.path.isfile(target))
+        self.assertEqual(toy_ec_txt, ft.read_file(target))
+
+        ft.remove_file(target)
+
+        # default behaviour is to copy single file list to target *directory*
+        self.assertFalse(os.path.exists(target))
+        ft.copy_files([toy_ec], target)
+        self.assertTrue(os.path.exists(target))
+        self.assertTrue(os.path.isdir(target))
+        copied_toy_ec = os.path.join(target, 'toy-0.0.eb')
+        self.assertTrue(os.path.exists(copied_toy_ec))
+        self.assertEqual(toy_ec_txt, ft.read_file(copied_toy_ec))
+
+        ft.remove_dir(target)
+
+        # test enabling verbose mode
+        self.mock_stderr(True)
+        self.mock_stdout(True)
+        ft.copy_files([toy_ec], target, verbose=True)
+        stderr, stdout = self.get_stderr(), self.get_stdout()
+        self.mock_stderr(False)
+        self.mock_stdout(False)
+        self.assertEqual(stderr, '')
+        regex = re.compile(r"^1 file\(s\) copied to .*/target")
+        self.assertTrue(regex.match(stdout), "Pattern '%s' should be found in: %s" % (regex.pattern, stdout))
+
+        ft.remove_dir(target)
+
+        self.mock_stderr(True)
+        self.mock_stdout(True)
+        ft.copy_files([toy_ec], target, target_single_file=True, verbose=True)
+        stderr, stdout = self.get_stderr(), self.get_stdout()
+        self.mock_stderr(False)
+        self.mock_stdout(False)
+        self.assertEqual(stderr, '')
+        regex = re.compile(r"^toy-0\.0\.eb copied to .*/target")
+        self.assertTrue(regex.match(stdout), "Pattern '%s' should be found in: %s" % (regex.pattern, stdout))
+
+        ft.remove_file(target)
+
+        # check behaviour under -x: only printing, no actual copying
+        init_config(build_options={'extended_dry_run': True})
+        self.assertFalse(os.path.exists(target))
+        self.assertFalse(os.path.exists(os.path.join(target, 'test.eb')))
```

Comment:
Isn't this somewhat redundant, if target doesn't exist (as per the assertFalse above), then target/test.eb can't exist